gpytorch/kernels/cosine_kernel.py

```diff
@@ -21,9 +20,9 @@ class CosineKernel(Kernel):
     where :math:`p` is the periord length parameter.
 
     Args:
-        :attr:`batch_size` (int, optional):
+        :attr:`batch_shape` (torch.Size, optional):
             Set this if you want a separate lengthscale for each
-            batch of input data. It should be `b` if :attr:`x1` is a `b x n x d` tensor. Default: `1`
+            batch of input data. It should be `b` if :attr:`x1` is a `b x n x d` tensor. Default: `torch.Size([1])`
```

Comment:
Can the default be `torch.Size([])`?

Reply:
Not without being breaking or just special casing that default to add a `[1]` anyways.
```python
torch.nn.Parameter(torch.zeros(*self.batch_shape, 1, lengthscale_num_dims)
```
Should by default resolve to `torch.zeros(1, 1, lengthscale_num_dims)`. 

Removing the default leading 1 is a much larger refactor that involves:
1. Not expanding `x` internally to be at least 3D
2. Dealing with the breaking change to model loading.

Since the goal of getting multibatch working doesn't require that, my preference would be to do a second PR for that refactor if we want it.

Reply:
I would say that we definitely want to do that. But you're right that it could be in a different PR.

We could do a PR that makes Likelihoods multibatch capable, and then do a PR that sets the default batch shape for both of them to `torch.Size([])` and fixes model loading.

Reply:
I think that the `torch.Size([])` change should be made before 0.3.0 for sure though.

Reply:
Yeah we can do it before 0.3, but I think it'll make the most sense to do it all at once for likelihoods and kernels so we don't have a window on master where lengthscales are `1` by default but noises are `1 x 1 x 1` by default.

Reply:
Sounds like a plan.